### PR TITLE
Task/MS2-331: Sport 1 - Player needs to get the stream token - iOS

### PR DESCRIPTION
**Pull request by TolyaAfanasev**
- getting stream token from API.

### CODE REVIEW

File: Sport1Player/Sport1PlayerAdapter.m [hunk 1]
```diff
@@ -318,24 +328,95 @@ -(void)shouldPresentPin {
...
++ (void)getStreamTokenFromAPI:(void (^)(BOOL success, NSString *streamToken))completion {
+
+    NSString *userToken = [[ZPLoginManager.sharedInstance createWithUserData] getUserToken];
+
+    if (!userToken) {
+        completion(NO, nil);
+        return;
+    }
+
+    NSURL *url = [NSURL URLWithString:@"https://services.inplayer.com/items/71614/access"];
```
Comment (PRuedaApplicaster) [inline Sport1Player/Sport1PlayerAdapter.m]: The identifier 71614 should be taken from the model. It is coming in the extensions dictionary under the key auth_id, and it is 2 nubmers separed by a _ (71614_66734), you should take just the first part.

File: Sport1Player/Sport1PlayerAdapter.m [hunk 2]
```diff
@@ -318,24 +328,95 @@ -(void)shouldPresentPin {
...
+    if (!current.isLive) {
+        completion(current);
+        return;
+    }
+
+    [Sport1PlayerAdapter getStreamTokenFromAPI:^(BOOL success, NSString *streamToken) {
+        if (!success || !streamToken) {
+            completion(current);
+            return;
+        }
```
Comment (PRuedaApplicaster) [inline Sport1Player/Sport1PlayerAdapter.m]: If the API returns a 401 or 403 we should launch the login again.

File: Sport1Player/Sport1PlayerAdapter.m (hunk 1 above)
Comment (TolyaAfanasev) [reply Sport1Player/Sport1PlayerAdapter.m]: Hi @PRuedaApplicaster , i'm trying to get it from (NSObject <ZPPlayable>* ) but it's empty.

Comment (TolyaAfanasev) [reply Sport1Player/Sport1PlayerAdapter.m]: There is only one value in the extensionsDictionary:
{
    free = 0;
}

Comment (PRuedaApplicaster) [reply Sport1Player/Sport1PlayerAdapter.m]: You can use the extensionsDictionary of the ZPPlayable object. Look the screenshot:
Screenshot 2019-10-24 at 11 55 10

Comment (PRuedaApplicaster) [reply Sport1Player/Sport1PlayerAdapter.m]: I am getting more values in the extensionsDictionary. What video are you using, or how are you testing it?

File: Sport1Player/Sport1PlayerAdapter.m (hunk 2 above)
Comment (TolyaAfanasev) [reply Sport1Player/Sport1PlayerAdapter.m]: Yes, i'm using exactly this object.
My test scenarion:
- open eSport1 app
- go to second Tab (Live)
- catch break point in method amendIfLivestreamModified

Comment (PRuedaApplicaster) [reply Sport1Player/Sport1PlayerAdapter.m]: We are having problems in that tab at the moment. Please select the first video from the home screen. It is also a livestream.

Comment (TolyaAfanasev) [reply Sport1Player/Sport1PlayerAdapter.m]: Ok, works fine with video from the first tab.

**Comment by TolyaAfanasev** [comment]
Hi @PRuedaApplicaster, could you please check my new changes ?

**Comment by TolyaAfanasev** [comment]
Hi @PRuedaApplicaster, yes you are right.
I thought we should invoke login (get stream token)... but now i got it - we need to invoke login screen, and get new token from there. Will do.

**Pull request merged**